lib/oli/delivery/sections.ex

```diff
@@ -621,6 +644,30 @@ defmodule Oli.Delivery.Sections do
     update_section(section, %{status: :deleted})
   end
 
+  @doc """
+  Archives a section by marking the record as archived.
+  ## Examples
+      iex> archive_section(section)
+      {:ok, %Section{}}
+      iex> archive_section(section)
+      {:error, %Ecto.Changeset{}}
+  """
+  def archive_section(%Section{} = section) do
```

Comment:
I feel like the update function called vs having a specific function just for this. Feels super redundant.